plugins/cilium-cni/chaining/awscni/network.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package awscni
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/linux_defaults"
+	"github.com/cilium/cilium/pkg/datapath/linux/route"
+)
+
+// awsCNIVLANIfacePrefix is the prefix used by the AWS CNI for building
+// interface names for SGP Pod VLAN interfaces
+const awsCNIVLANIfacePrefix = "vlan.eth."
+
+// buildSGPPodVLANIfaceName returns the name for the Pod associated with
+// the VLAN ID vlanID
+func buildSGPPodVLANIfaceName(vlanID string) string {
+	return awsCNIVLANIfacePrefix + vlanID
+}
+
+// disableIfaceRPFilter disables the `rp_filter` network setting for the
+// network interface iface
+func disableIfaceRPFilter(iface string) error {
+	return os.WriteFile(
+		fmt.Sprintf("/proc/sys/net/ipv4/conf/%s/rp_filter", iface),
+		[]byte("0"),
+		0644)
```

Comment:
nit: Could we use `sysctl.Write` here? Here's an example of how this can be done from the CNI plugin: https://github.com/cilium/cilium/blob/82a8f409acc6bee8ce71f444ac1d892d8775383b/plugins/cilium-cni/cmd/cmd.go#L424

Reply:
I second this, its preferred reuse the same API whenever we can. Since the CNI isn't in the agent you can't use the reconciling sysctl logic, but we have a fallback available: https://github.com/cilium/cilium/blob/d49bec4b6e047d55b01efeffdad18a338907936e/pkg/datapath/linux/sysctl/sysctl.go#L181-L196

We even have an existing copy of this very logic on tap: https://github.com/cilium/cilium/blob/d49bec4b6e047d55b01efeffdad18a338907936e/pkg/datapath/connector/add.go#L48

Reply:
Oh, I only now noticed you fixed this in a later commit (I am reviewing one by one). Can you rebase this old commit instead so we keep a clean git history?